api/v1/stats/PKPStatsHandler.inc.php

```diff
@@ -247,15 +252,120 @@ public function getSubmission($slimRequest, $response, $args) {
+			->getFullProperties($submission, [
 				'request' => $request,
-				'slimRequest' 	=> $slimRequest,
+				'slimRequest' => $slimRequest,
 				'params' => $params
-			));
+			]);
 
 		return $response->withJson($data, 200);
 	}
 
+	/**
+	 * Retrieve an overview of all the active submissions, including yearly statistics
+	 * @param $slimRequest object Request Slim request
+	 * @param $response object Response
+	 * @param $args array
+	 * @return object Response
+	 */
+	public function getEditorialReport(\Slim\Http\Request $slimRequest, APIResponse $response, $args) {
+		$request = \Application::getRequest();
+		$context = $request->getContext();
+
+		if (!$context) {
+			return $response->withStatus(404)->withJsonError('api.submissions.404.resourceNotFound');
+		}
+
+		AppLocale::requireComponents(
+			LOCALE_COMPONENT_PKP_USER,
+			LOCALE_COMPONENT_PKP_MANAGER,
+			LOCALE_COMPONENT_PKP_SUBMISSION
+		);
+
+		// Convert params passed to the api end point
+		// Merge query params over default params
+		$defaultParams = ['timeSegment' => 'year'];
+		$requestParams = $slimRequest->getQueryParams() + $defaultParams;
+
+		$params = [];
+		// Process query params to format incoming data as needed
+		foreach ($requestParams as $param => $val) {
+			switch ($param) {
+				case 'timeSegment':
+				case 'dateStart':
+				case 'dateEnd':
+					$params[$param] = $val;
+					break;
+				case 'sectionIds':
+					// these are section IDs in OJS and series IDs in OMP
+					if (is_string($val) && strpos($val, ',') > -1) {
+						$val = explode(',', $val);
+					} elseif (!is_array($val)) {
+						$val = [$val];
+					}
+					$params[$param] = array_map('intval', $val);
+					break;
+			}
+		}
+
+		\HookRegistry::call('API::stats::editorialReport::params', [&$params, $slimRequest]);
+
+		// validate parameters
+		if (isset($params['dateStart'])) {
+			if (!$this->validDate($params['dateStart'])) {
+				return $response->withStatus(400)->withJsonError('api.stats.400.wrongDateFormat');
+			}
+		}
+		if (isset($params['dateEnd'])) {
+			if (!$this->validDate($params['dateEnd'])) {
+				return $response->withStatus(400)->withJsonError('api.stats.400.wrongDateFormat');
+			}
+		}
+		if (isset($params['dateStart']) && isset($params['dateEnd'])) {
+			if (!$this->validDateRange($params['dateStart'], $params['dateEnd'])) {
+				return $response->withStatus(400)->withJsonError('api.stats.400.wrongDateRange');
+			}
+		}
+
+		import('lib.pkp.controllers.stats.EditorialReportComponentHandler');
+
+		$paramsWithoutDateRange = array_diff_key($params, ['dateStart' => null, 'dateEnd' => '']);
+
+		$statsService = \ServicesContainer::instance()->get('stats');
+		$statistics = $statsService->getSubmissionStatistics($context->getId(), $paramsWithoutDateRange);
+		$rangedStatistics = $statsService->getSubmissionStatistics($context->getId(), $params);
+
+		$submissionChartData = EditorialReportComponentHandler::extractSubmissionChartData($statistics);
```

Comment:
I've left these extract* functions in the `EditorialReportComponentHandler` because they are only "template fillers", they carry no importance/logic whatsoever... I've turned them into functions only to avoid duplicating code.

Since it's going to disappear anyway, due to the structural modifications, I'll move them to the `EditorialStatisticsService` =]